src/radical/saga/task.py

```diff
@@ -94,6 +94,8 @@ def __init__ (self, _adaptor, _method_type, _method_context, _ttype) :
         self._attributes_set_getter(c.STATE,     self.get_state)
         self._attributes_set_setter(c.STATE,     self._set_state)
 
+        self._uid = ru.generate_id('task.')
```

Comment:
`generate_id` already creates ID for Task (from inherited `rs.SimpleBase`, [link](https://github.com/radical-cybertools/radical.saga/blob/devel/src/radical/saga/base.py#L47)) -> `self._id`

Reply:
Good catch, thanks!